### PR TITLE
only publish images if image tests run

### LINKED ISSUE #10968: sometimes visual testing fail with `no such file or directory`
Take the example of https://github.com/Qiskit/qiskit/pull/10907/checks?check_run_id=17192520206:

screencapture-github-Qiskit-qiskit-pull-10907-checks-2023-10-04-14_24_38

"Sometimes" (maybe there is a pattern, I just dont know which one) this happens. No idea why.

### PR DESCRIPTION
**Pull request by AngeloDanducci**
### Summary
Fixes #10968

### Details and comments
Need to open PR to verify CI tests work correctly, fixes the flake of too early a failure causing image publishing to fail

### CODE REVIEW

**Comment by AngeloDanducci** [comment]
Will re convert after CI has verified correctly I've removed temporary test changes.

**Review by 1ucian0 — DISMISSED** [review]
Thanks a lot for the fix!

File: .azure/test-linux.yml [hunk 1]
```diff
@@ -197,14 +198,16 @@ jobs:
...
+            echo "##vso[task.setvariable variable=HAVE_VISUAL_TESTS_RUN;]true"
+            image_tests/bin/python -m unittest discover -v test/visual
           displayName: 'Run image test'
           env:
             # Needed to suppress a warning in jupyter-core 5.x by eagerly migrating to
             # a new internal interface that will be the default in jupyter-core 6.x.
             # This variable should become redundant on release of jupyter-core 6.
             JUPYTER_PLATFORM_DIRS: 1
-
+
```
Comment (jakelishman) [inline .azure/test-linux.yml]:
```suggestion

```

File: .azure/test-linux.yml [hunk 2]
```diff
@@ -213,7 +216,7 @@ jobs:
             archiveType: tar
             archiveFile: '$(Build.ArtifactStagingDirectory)/visual_test_failures.tar.gz'
             verbose: true
-          condition: failed()
+          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```
Comment (jakelishman) [inline .azure/test-linux.yml]:
```suggestion
          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```

File: .azure/test-linux.yml [hunk 3]
```diff
@@ -222,7 +225,7 @@ jobs:
             archiveType: tar
             archiveFile: '$(Build.ArtifactStagingDirectory)/circuit_results.tar.gz'
             verbose: true
-          condition: failed()
+          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```
Comment (jakelishman) [inline .azure/test-linux.yml]:
```suggestion
          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```

File: .azure/test-linux.yml [hunk 4]
```diff
@@ -231,7 +234,7 @@ jobs:
             archiveType: tar
             archiveFile: '$(Build.ArtifactStagingDirectory)/graph_results.tar.gz'
             verbose: true
-          condition: failed()
+          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```
Comment (jakelishman) [inline .azure/test-linux.yml]:
```suggestion
          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```

File: .azure/test-linux.yml [hunk 5]
```diff
@@ -240,4 +243,4 @@ jobs:
             artifactName: 'image_test_failure_img_diffs'
             Parallel: true
             ParallelCount: 8
-          condition: failed()
+          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```
Comment (jakelishman) [inline .azure/test-linux.yml]:
```suggestion
          condition: and(failed(), eq(variables.HAVE_VISUAL_TESTS_RUN, 'true'))
```

**Review by jakelishman — COMMENTED** [review]
Thanks for this - this looks like the kind of solution I was envisaging!

Minor: please could we tidy up the trailing whitespace?

**Review by jakelishman — APPROVED** [review]
Thanks!

**Pull request merged**